### PR TITLE
Bump version numbers and add upgrade guide for the 2.9.0 release

### LINKED ISSUE #571: Documentation update for 2.9.0
This is a tracking ticket for docs related to the upcoming 2.9.0 release of SecureDrop

### PR DESCRIPTION
**Pull request by nathandyer**
## Status

Ready for review

## Description of Changes

This PR bumps the version numbers throughout the documentation to 2.9.0, and adds an upgrade guide for this release.

I left the instructions for the Tails 6 upgrade in the guide since some folks may not have completed the upgrade as part of the previous release. If we feel these are not necessary I'm happy to remove them. Just let me know!

* Resolves #571

## Testing
- [ ] Visual review

## Release 
- [ ] We will need to tag a 2.9.0 stable docs release once 2.9.0 is out the door

## Checklist (Optional)

- [x] Doc linting (`make docs-lint`) passed locally
- [ ] Doc link linting (`make docs-linkcheck`) passed
- [x] You have previewed (`make docs`) docs at http://localhost:8000

### CODE REVIEW

File: docs/upgrade/2.8.0_to_2.9.0.rst
Comment (eloquence) [inline docs/upgrade/2.8.0_to_2.9.0.rst:15]: Maybe we should add "If you have already upgraded to Tails 6, you only need to complete Step 1 below."

File: docs/index.rst
Comment (eloquence) [inline docs/index.rst:154]: I'd suggest removing the oldest guide while we're here :)

**Comment by eloquence** [comment]
LGTM, two nits inline. I think keeping the Tails 6 docs is a good idea for now; if we want to keep them a few months longer, it might be worth reorganizing them into their own section instead of copying them with each upgrade guide. But removing them with 2.10.0 is probably fine.

File: docs/index.rst
Comment (nathandyer) [reply docs/index.rst:154]: Good idea!

**Comment by nathandyer** [comment]
Thanks @eloquence, that sounds great. I think we're probably okay to remove them in 2.10.0, but we can decide one way or the other closer to that release once we have a better feel for how widely Tails 6 has been adopted.

**Review by eloquence — APPROVED** [review]
Good to go once release is  live.

**Comment by nathandyer** [comment]
Thanks! After discussing with @zenmonkeykstop , I added an additional commit which mentions the new Tor PoW option, and links to our guide for enabling it.

Will need another review before it can be merged with the release.

**Review by legoktm — APPROVED** [review]
I only reviewed the most recent commit (the Tor PoW notice), and that LGTM. Approving based on Erik's prior approval.

**Pull request merged**
